src/main/kotlin/de/docs_as_co/intellij/plugin/drawio/editor/DiagramsEditorProvider.kt

```diff
@@ -0,0 +1,39 @@
+package de.docs_as_co.intellij.plugin.drawio.editor
+
+import com.intellij.openapi.application.Application
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.command.WriteCommandAction
+import com.intellij.openapi.fileEditor.*
+import com.intellij.openapi.project.DumbAware
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.Key
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.testFramework.runNonUndoableWriteAction
+import com.jetbrains.rd.util.lifetime.LifetimeDefinition
+import java.beans.PropertyChangeListener
+import javax.swing.JComponent
+
+class DiagramsEditorProvider : FileEditorProvider, DumbAware {
+    /**
+     * accept is called whenever IntelliJ opens an editor
+     * if accept return true, IntelliJ will open an instance of this editor
+     */
+    override fun accept(project: Project, file: VirtualFile): Boolean {
+        if (file.isDirectory || !file.exists()) {
+            return false;
+        }
+        val extensions = "drawio;drawio.svg;dio;dio.svg"
+        extensions.split(";").forEach() {
+            if (file.name.endsWith("."+it)) {
+                return true
+            }
+        }
```

Comment:
I think this might be a little bit more readable:
```kotlin
val extensions = arrayOf(".drawio", ".drawio.svg", ".dio", ".dio.svg")
if (extensions.some(file.name.endsWith)) {
    return true
}
```

Reply:
LOL. I guess it depends.
I used the split because I wanted to have a file extension list in the ressources in order to have it only once in the code. But I didn't manage to use a Ressource in the `plugin.xml`

the next statement would look in Groovy something like this:

    if (extensions.find { file.name.endsWith(it) } ) return true

I guess we can do the same in Kotlin, can't we?

    if (extensions.some( file.name.endsWith (it) ) return true

btw: wouldn't be `.any` a better choice then `.some`?

Reply:
btw - I can't find the `.some` method, only `.any`

this seems to be a code line which works:

        if (extensions.any { ext -> file.name.endsWith(ext)}) { 
            return true 
        }

Reply:
> But I didn't manage to use a Ressource in the plugin.xml

If you do manage to do this, I think it's a good idea.
And yeah, it is `some`, not `any`.

This:
```kotlin
if (extensions.any { ext -> file.name.endsWith(ext) }) { 
    return true 
}
```
should be equivalent to this:
```kotlin
if (extensions.any(file.name.endsWith)) { 
    return true 
}
```
But it is totally ok for me to not do this [eta-conversion](https://sookocheff.com/post/fp/eta-conversion/) ;)

Reply:
refactored